app/models/piece.rb

```diff
@@ -18,12 +17,47 @@ def move!(x, y)
     true
   end
 
+  # Return false if coordinates is taken but belongs to friendly piece
+  # otherwise return true
+  def capturable?(x, y)
+    return false unless position_taken?(x, y) && !friendly_piece?(x, y)
+    true
+  end
+
+  # if piece is capturable change piece's attributes
+  def capture!(x, y)
+    return false unless capturable?(x, y)
+    move!(x, y)
+    target_piece(x, y).update_attributes!(
+      x_position: nil,
+      y_position: nil,
+      captured: true
+    )
+  end
```

Comment:
I think you are misunderstanding what I'm saying.  Move should initiate the capture method.  Move is the action that is cascading all these other actions.  The logic for move method is literally 1 line of code right now.  Adding a capturable condition to it that triggers capture will make it 2 lines of code.  The whole point is to have everything build into move in a way that is clear and transparent.